core/rawdb/chain_iterator.go

```diff
@@ -58,26 +58,23 @@ func InitDatabaseFromFreezer(db ethdb.Database) {
 			number := i + uint64(j)
 			hash = common.BytesToHash(h)
 			WriteHeaderNumber(batch, hash, number)
+
 			// If enough data was accumulated in memory or we're at the last block, dump to disk
-			if batch.ValueSize() > ethdb.IdealBatchSize {
+			if batch.ValueSize() > ethdb.IdealBatchSize || j == len(data)-1 {
```

Comment:
This trick shaves off three lines of code, true, but it IMO makes it a bit brittle, e.g if we later add a `break` in the loop. 

In a lot++ of places in the code, we do a batch-write after the loop to clear out any stragglers, I see no need to fix that personally.

Reply:
reverted.